ts/output/common/FontData.ts

```diff
@@ -673,27 +804,63 @@ export class FontData<C extends CharOptions, V extends VariantData<C>, D extends
+    let CLASS = this.CLASS;
     this.options = userOptions(defaultOptions({}, CLASS.OPTIONS), options);
     this.params = {...CLASS.defaultParams};
     this.sizeVariants = [...CLASS.defaultSizeVariants];
     this.stretchVariants = [...CLASS.defaultStretchVariants];
-    this.cssFontMap = {...CLASS.defaultCssFonts};
-    for (const name of Object.keys(this.cssFontMap)) {
-      if (this.cssFontMap[name][0] === 'unknown') {
-        this.cssFontMap[name][0] = this.options.unknownFamily;
-      }
-    }
+    this.defineCssFonts(CLASS.defaultCssFonts);
     this.cssFamilyPrefix = CLASS.defaultCssFamilyPrefix;
     this.createVariants(CLASS.defaultVariants);
-    this.defineDelimiters(CLASS.defaultDelimiters);
-    for (const name of Object.keys(CLASS.defaultChars)) {
-      this.defineChars(name, CLASS.defaultChars[name]);
-    }
+    this.defineDelimiters(CLASS.defaultDelimiters as DelimiterMap<D>);
+    Object.keys(CLASS.defaultChars).forEach(name => this.defineChars(name, CLASS.defaultChars[name] as CharMap<C>));
     this.defineRemap('accent', CLASS.defaultAccentMap);
     this.defineRemap('mo', CLASS.defaultMoMap);
     this.defineRemap('mn', CLASS.defaultMnMap);
     this.defineDynamicCharacters(CLASS.dynamicFiles);
+    CLASS.dynamicExtensions.forEach(data => this.defineDynamicCharacters(data.files));
+  }
+
+  /**
+   * Add an extension to an existing font instance (options will get their defaults).
+   *
+   * @param {FontExtensionData} data    The data for the font extension to merge into this font.
+   * @param {string} prefix             The [prefix] to add to all component names
+   */
+  public addExtension(data: FontExtensionData<C, D>, prefix: string = '') {
+    const dynamicFont = {
+      name: data.name,
+      prefix: prefix,
+      files: this.CLASS.defineDynamicFiles(data.ranges, prefix),
+      sizeN: this.sizeVariants.length,
+      stretchN: this.stretchVariants.length
+    };
+    this.CLASS.dynamicExtensions.set(data.name, dynamicFont);
+
+    data.options && defaultOptions(this.options, data.options);
+    data.parameters && defaultOptions(this.params, data.parameters);
+    if (data.sizeVariants) {
+      this.sizeVariants = mergeOptions(this.sizeVariants, data.sizeVariants);
+    }
+    if (data.stretchVariants) {
+      this.stretchVariants = mergeOptions(this.stretchVariants, data.stretchVariants);
+    }
+    data.cssFonts && this.defineCssFonts(mergeOptions([], data.cssFonts));
+    data.variants && this.createVariants(mergeOptions([], data.variants));
+    if (data.delimiters) {
+      this.defineDelimiters(mergeOptions([], data.delimiters));
+      this.CLASS.adjustDelimiters(this.delimiters, Object.keys(data.delimiters),
+                                  dynamicFont.sizeN, dynamicFont.stretchN);
+    }
+    for (const name of Object.keys(data.chars || {})) {
+      this.defineChars(name, data.chars[name]);
+    }
+    data.accentMap && this.defineRemap('accent', data.accentMap);
```

Comment:
Same comment for `defineRemap` as for ` mergeOptions`.

Reply:
Also changed.